components/dashboard/src/utils.ts

```diff
@@ -213,3 +213,22 @@ export class ReplayableEventEmitter<EventTypes extends EventMap> extends EventEm
         return this.reachedEnd;
     }
 }
+
+export function isTrustedUrlOrPath(urlOrPath: string) {
+    try {
+        const ok = URL.canParse(urlOrPath);
+        if (ok) {
+            const u = new URL(urlOrPath);
+            const isTrusted = window.location.hostname === u.hostname;
+            if (!isTrusted) {
+                console.warn("Untrusted URL", urlOrPath);
+            }
+            return isTrusted;
+        } else {
+            return urlOrPath.startsWith("/");
```

Comment:
For instance, for the CLI we use paths without the leading slash

```
>> urlOrPath
"api/oauth/authorize?client_id=gitpod-cli&code_challenge=2R902KlBwwAYC44_WtF5CxNuejV7zaiSqq6msC_EQkA&code_challenge_method=S256&redirect_uri=http%3A%2F%2F127.0.0.1%3A63110&response_type=code&scope=function%3AlistenForWorkspaceInstanceUpdates+function%3AgetGitpodTokenScopes+function%3AgetLoggedInUser+function%3AaccessCodeSyncStorage+function%3AgetOwnerToken+function%3AgetWorkspace+function%3AgetWorkspaces+function%3AgetSSHPublicKeys+function%3AstartWorkspace+function%3AstopWorkspace+function%3AdeleteWorkspace+function%3AgetTeam+function%3AgetTeams+function%3AgetTeamMembers+function%3AgetTeamProjects+function%3AcreateWorkspace+function%3AgetToken+function%3AgetSupportedWorkspaceClasses+function%3AgetIDEOptions+resource%3Adefault&state=state&approved=yes"
>> urlOrPath.startsWith("/")
false
>> URL.canParse(urlOrPath)
false 
```

Reply:
👍

Reply:
Addressed https://github.com/gitpod-io/gitpod/pull/20152/commits/edfabf88a382e75eae93f5b0ff04ae6902f994e2